src/clj/witan/send/goal_seek.clj

```diff
@@ -25,35 +31,40 @@
                       (update :population i/->int)))
             filepath))
 
-(defn state-pop
-  "Takes a config and returns the state to be modified and output path of
-   Output_State_pop_only.csv results"
-  [config]
-  (let [state (-> (get-in config [:transition-parameters :transitions-to-change])
-                  first
-                  (clojure.set/rename-keys {:setting-2 :setting :need-2 :need :academic-year-2 :academic-year})
-                  (select-keys [:setting :need :academic-year]))
-        out-dir (:output-dir (:output-parameters config))
-        state-pop (str (:project-dir config)
-                       "/"
-                       out-dir
-                       "/Output_State_pop_only.csv")]
-    [state state-pop]))
+(defn match-need-setting [data state]
+  (= (second (key data)) (st/join-need-setting (:need state) (:setting state))))
+
+(defn match-ay [data state]
+  (let [ay-state (:academic-year state)]
+    (if (coll? ay-state)
+      (true? (some (fn [ay] (= (first (key data)) ay)) ay-state))
+      (= (first (key data)) ay-state))))
 
 (defn get-target-pop
   "Takes a vector with a state to find (e.g. {:setting :SFSS, :need :ASD}) and filepath
   for a Output_State_pop_only.csv file to return summed populations sorted by calender year
   for ease of reading in REPL"
-  [[state state-pop]]
-  (->> (csv->state-pop state-pop)
-       (filter #(every? identity (p/test-predicates % state))) ;; need something to build predicates
-       (group-by :calendar-year)
-       (map #(merge (assoc {} :year (key %))
-                    (apply merge-with + (map (fn [m] (select-keys m [:population])) (val %)))))
-       (sort-by :year)))
+  [config state-pop]
+  (let [state (-> (get-in config [:transition-parameters :transitions-to-change])
+                  first
+                  (clojure.set/rename-keys {:setting-2 :setting :need-2 :need :academic-year-2 :academic-year})
+                  (select-keys [:setting :need :academic-year]))]
+    (->> state-pop
+         (filter #(cond (every? (fn [k] (contains? state k)) [:setting :need :academic-year])
+                        (and (match-need-setting % state)
+                             (match-ay % state))
+
+                        (every? (fn [k] (contains? state k)) [:setting :need])
+                        (match-need-setting % state)
 
-(defn update-results-path [config]
-  (assoc-in config [:output-parameters :output-dir] (str "results/" (get-in config [:output-parameters :output-dir]))))
+                        (contains? state :academic-year)
+                        (match-ay % state)))
+         (map val)
+         (apply merge-with +)
+         :median)))
```

Comment:
I presume at some point this will need to be a real median?

Reply:
ah are you pointing out that I'm summing medians...

Reply:
yes

Reply:
yes, then, although for this purpose, providing the previos projection has been run 1000 times, this is sufficient, as it's just used as a check and basic to start the modifier from